roles/openshift_node_group/files/sync.yaml

```diff
@@ -126,8 +129,25 @@ spec:
             md5sum /etc/origin/node/tmp/node-config.yaml > /tmp/.new
             if [[ "$( cat /tmp/.old )" != "$( cat /tmp/.new )" ]]; then
               mv /etc/origin/node/tmp/node-config.yaml /etc/origin/node/node-config.yaml
+              NODERESTART=1
+              echo "info: Node configuration changed, mark kubelet for restart" 2>&1
+            fi
+
+            # detect whether the volume-config.yaml has changed, and if so trigger a restart of the kubelet.
+            if [[ ! -f /etc/origin/node/volume-config.yaml ]]; then
+              cat /dev/null > /tmp/.vol-old
+            fi
+
+            md5sum /etc/origin/node/tmp/volume-config.yaml > /tmp/.vol-new
```

Comment:
This will cause an error if volume-config.yaml is not present (which is valid) and the sync pod will terminate. Consider adding a check to see if volume-config.yaml exists.

Reply:
Sounds good.